api/v1beta1/hetznerbaremetalhost_types.go

```diff
@@ -598,7 +597,7 @@ func (host *HetznerBareMetalHost) ClearRebootAnnotations() {
 }
 
 func isRebootAnnotation(annotation string) bool {
-	return strings.HasPrefix(annotation, RebootAnnotation+"/") || annotation == RebootAnnotation
+	return annotation == RebootAnnotation
```

Comment:
@janiskemper I think this change is safe. Or can there be different Annotations?